File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    honor = []
    for i in score:
        if len(honor) < k:
            honor.append(i)
            honor.sort(reverse = True)
            answer.append(honor[len(honor)-1])
        else:
            if honor[len(honor)-1] < i:
                honor[len(honor)-1] = i
            honor.sort(reverse = True)
            answer.append(honor[len(honor)-1])
    return answer

File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    const answer = [];
    const honor = [];
    
    for(let i=0; i < score.length; i++){
        if(honor.length < k) {
            honor.push(score[i]);
            answer.push(Math.min(...honor));
        }
        else {
            honor.sort((a,b) => b-a);
            if(honor.at(-1) < score[i]){
                honor.pop();
                honor.push(score[i]);
            };
            answer.push(Math.min(...honor));
        }
    }
    return answer;
}